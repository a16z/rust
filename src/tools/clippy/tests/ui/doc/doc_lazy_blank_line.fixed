// https://github.com/rust-lang/rust-clippy/issues/12917
#![warn(clippy::doc_lazy_continuation)]

/// This is a constant.
///
/// The meaning of which should not be explained.
pub const A: i32 = 42;

/// This is another constant, no longer used.
///
/// This block of documentation has a long
/// explanation and derivation to explain
/// why it is what it is, and how it's used.
///
/// It is left here for historical reasons, and
/// for reference.
///
/// Reasons it's great:
///  - First reason
///  - Second reason
///
//pub const B: i32 = 1337;

/// This is yet another constant.
///
/// This has a similar fate as `B`.
///
/// Reasons it's useful:
///  1. First reason
///  2. Second reason
///
//pub const C: i32 = 8008;

/// This is still in use.
pub const D: i32 = 20;

/// > blockquote code path
///

/// bottom text
pub const E: i32 = 20;

/// > blockquote code path
///
#[repr(C)]
/// bottom text
pub struct Foo(i32);
