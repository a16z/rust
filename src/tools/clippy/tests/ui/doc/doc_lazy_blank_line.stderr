error: doc list item without indentation
  --> tests/ui/doc/doc_lazy_blank_line.rs:23:5
   |
LL | /// This is yet another constant.
   |     ^
   |
   = help: if this is intended to be part of the list, indent 3 spaces
   = note: `-D clippy::doc-lazy-continuation` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::doc_lazy_continuation)]`
help: if this should be its own paragraph, add a blank doc comment line
   |
LL ~ ///  - Second reason
LL + ///
   |

error: doc list item without indentation
  --> tests/ui/doc/doc_lazy_blank_line.rs:32:5
   |
LL | /// This is still in use.
   |     ^
   |
   = help: if this is intended to be part of the list, indent 4 spaces
help: if this should be its own paragraph, add a blank doc comment line
   |
LL ~ ///  2. Second reason
LL + ///
   |

error: doc quote line without `>` marker
  --> tests/ui/doc/doc_lazy_blank_line.rs:37:5
   |
LL | /// bottom text
   |     ^
   |
   = help: if this not intended to be a quote at all, escape it with `\>`
help: if this should be its own paragraph, add a blank doc comment line
   |
LL ~ /// > blockquote code path
LL + ///
   |

error: doc quote line without `>` marker
  --> tests/ui/doc/doc_lazy_blank_line.rs:42:5
   |
LL | /// bottom text
   |     ^
   |
   = help: if this not intended to be a quote at all, escape it with `\>`
help: if this should be its own paragraph, add a blank doc comment line
   |
LL ~ /// > blockquote code path
LL + ///
   |

error: aborting due to 4 previous errors

